src/Microsoft.Data/DataFrame.cs

```diff
@@ -133,13 +133,124 @@ public DataFrame Sort(string columnName, bool ascending = true)
             for (int i = 0; i < ColumnCount; i++)
             {
                 BaseColumn oldColumn = Column(i);
-                BaseColumn newColumn = oldColumn.CloneAndAppendNulls(sortIndices, !ascending);
+                BaseColumn newColumn = oldColumn.CloneAndAppendNulls(oldColumn.NullCount, sortIndices, !ascending);
                 Debug.Assert(newColumn.NullCount == oldColumn.NullCount);
                 newColumns.Add(newColumn);
             }
             return new DataFrame(newColumns);
         }
 
+        public enum JoinAlgorithm
+        {
+            Left,
+            Right,
+            Outer,
+            Inner
+        }
+
+        public DataFrame Join(DataFrame other, string leftSuffix = "_left", string rightSuffix = "_right", JoinAlgorithm joinAlgorithm = JoinAlgorithm.Left)
```

Comment:
I see in Pandas, the default is blank for left and right suffix. Do we always want a suffix?

Reply:
Yup. Because pandas will fail if you try to join without the suffix and there are duplicate column names

Reply:
Done

Reply:
Do we want to match the Pandas behavior of throwing if there are conflicts by default? Or is defaulting to `_left` and `_right` suffixes better behavior?

Reply:
Personally I think defaulting to `_left` and `_right` is better behavior. I don't have a strong opinion about this though. It just feels like a place where we can do better than throwing an exception. I did however just realize that I had to change the `if` to a `while` though.